examples/fundamentals.test.js

```diff
@@ -4418,6 +4418,42 @@ describe('Fundamentals', function() {
       arr.findIndex(n => Number.isNaN(n)); // 4
       
     });
+    it('bad-undefined-check', function() {
+      // acquit:ignore:start
+      assert.throws(() => {
+      // acquit:ignore:end
+       x === undefined; // Throws a ReferenceError
+       typeof x == 'undefined'; // true
+      // acquit:ignore:start
+      }, /ReferenceError: x is not defined/);
+      // acquit:ignore:end
+    });
+    it('undefined-objects', function() {
+      // acquit:ignore:start
+      assert.throws(() => {
+      // acquit:ignore:end
+        var person = { firstName:"John", lastName:"Doe", age:50, eyeColor:"blue", test: undefined };
+
+        person.test == undefined; // true
+
+        ({})._does_not_exist == undefined; // true
+
+        person._does_not_exist === undefined; // true
+
+        person.hasOwnProperty(_does_not_exist); // ReferenceError
```

Comment:
This should be `person.hasOwnProperty('_does_not_exist')`, notice that the property name must be a string.